### PR TITLE
FAI-811: Add automatic initialization as default behavior

### PR DESCRIPTION
**Pull request by RobGeada**
[FAI-811](https://issues.redhat.com/browse/FAI-811)

Importing any trustyai submodule, such as `trustyai.model` or `trustyai.explainers` will now automatically initialize the jvm if it has not yet been initialized _in the current process_. However, `trustyai.initializer.init()` can also be called manually, which has the same behavior as the old `trustyai.init` function.

### CODE REVIEW

File: trustyai/__init__.py [hunk 1]
```diff
@@ -1,118 +1,15 @@
...
 if os.getenv("PYTHON_TRUSTY_DEBUG") == "1":
-    _logging_level = logging.DEBUG
+    _LOGGING_LEVEL = logging.DEBUG
 else:
-    _logging_level = logging.WARN
-
-logging.basicConfig(level=_logging_level)
-
-
-def init(*args, path=CORE_DEPS):
```
Comment (danielezonca) [inline trustyai/__init__.py:61]: Do we want to preserve `init()` as deprecated method to avoid a breaking change?
I was thinking at a empty `init()` that just warn the user to remove it because not necessary.

Wdyt?

**Review by tteofili — APPROVED** [review]
LGTM very nice thanks @RobGeada

File: trustyai/explainers.py [hunk 2]
```diff
@@ -403,16 +407,17 @@ def as_dataframe(self) -> pd.DataFrame:
 
         Returns
         -------
-        pandas.DataFrame
-            DataFrame containing the results of the SHAP explanation. For each model output,
+        Dict[String, pandas.DataFrame]
+            Dictionary of dataFrame containing the results of the SHAP explanation, keyed by the
+             model output named. For each model output,
```
Comment (danielezonca) [inline trustyai/explainers.py:412]: Is this change related? It seems a leftover from a different branch/pr

**Review by danielezonca — APPROVED** [review]

File: trustyai/explainers.py (hunk 2 above)
Comment (RobGeada) [reply trustyai/explainers.py:412]: unrelated, must have snuck in

File: trustyai/__init__.py (hunk 1 above)
Comment (RobGeada) [reply trustyai/__init__.py:61]: will do

**Review by ruivieira — APPROVED** [review]
LGTM, thanks @RobGeada.

I'm just wondering why storing the initialisation status with an environment variable?
Wouldn't it be easier to store it in a top-level variable such as `TRUSTYAI_IS_INITIALIZED = True / False`?
